### PR TITLE
Lt 4576 NotificationsService - Add Boilerplate code

### PR DESCRIPTION
**Pull request by fatihdumanli**
[LT-4576](https://lykke-snow.atlassian.net/browse/LT-4576)

This PR adds boilerplate code to the repository, by following the guidelines **New service checklist**
The .NET projects are created from the template https://github.com/LykkeBusinessPlatform/lykke.dotnettemplates

[LT-4576]: https://lykke-snow.atlassian.net/browse/LT-4576?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/Lykke.Snow.Notifications.Client/Lykke.Snow.Notifications.Client.csproj [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <DocumentationFile>$(OutputPath)$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Remove="Lykke.Snow.Notifications.Client.xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Lykke.HttpClientGenerator" Version="2.2.0" />
+    <PackageReference Include="Lykke.SettingsReader" Version="4.0.1" />
```
Comment (tarurar) [inline client/Lykke.Snow.Notifications.Client/Lykke.Snow.Notifications.Client.csproj:32]: I think we have newer versions already.

File: src/Lykke.Snow.Notifications/appsettings.dev.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (tarurar) [inline src/Lykke.Snow.Notifications/appsettings.dev.json:1]: Why do you have to post dev json settings into cvs?

File: tests/Lykke.Snow.Notifications.Tests/Lykke.Snow.Notifications.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <LangVersion>8.0</LangVersion>
+    <Version>1.0.0</Version>
+    <WarningsAsErrors>8597,8600,8601,8602,8603,8604,8605,8607,8608,8609,8610,8611,8612,8613,8614,8615,8616,8617,8618,8619,8620,8621,8622,8624,8625,8629,8631,8633,8634,8643,8644,8645,8655,8667,8670,8714,8762,8763,8764,8765,8766,8767,8768,8769,8770,8774,8775,8776,8777,8819,8824,8825,8847</WarningsAsErrors>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
```
Comment (tarurar) [inline tests/Lykke.Snow.Notifications.Tests/Lykke.Snow.Notifications.Tests.csproj:9]: I think package version have to be updated.

File: client/Lykke.Snow.Notifications.Client/INotificationsServiceClient.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using JetBrains.Annotations;
+
+namespace Lykke.Snow.Notifications.Client
+{
+    /// <summary>
+    /// Notifications API client interface.
+    /// </summary>
+    [PublicAPI]
+    public interface INotificationsServiceClient
```
[the remark this answers is not in the document]
Comment (fatihdumanli) [reply client/Lykke.Snow.Notifications.Client/INotificationsServiceClient.cs]: Removed

File: client/Lykke.Snow.Notifications.Client/Lykke.Snow.Notifications.Client.csproj (hunk 1 above)
Comment (fatihdumanli) [reply client/Lykke.Snow.Notifications.Client/Lykke.Snow.Notifications.Client.csproj]: Upgraded

File: settings.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+NotificationsService:
```
[the remark this answers is not in the document]
Comment (fatihdumanli) [reply settings.yaml]: Removed

File: src/Lykke.Snow.Notifications/Program.cs [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+        public static async Task Main(string[] args)
+        {
+            Log.Logger = new LoggerConfiguration()
+                .MinimumLevel.Debug()
+                .WriteTo.Console()
+                .WriteTo.File("logs/service.start.log")
+                .CreateBootstrapLogger();
+
+            var restartAttemptsLeft = int.TryParse(Environment.GetEnvironmentVariable("RESTART_ATTEMPTS_NUMBER"),
```
[the remark this answers is not in the document]
Comment (fatihdumanli) [reply src/Lykke.Snow.Notifications/Program.cs]: Used the Startup lib to implement retries https://github.com/LykkeBusiness/Lykke.Snow.Notifications/pull/1/files#diff-ea4570828d29a23295d2c2d7322ec1ae73033adc35f0261e13fc8ca3b91be2f4R17

Also, I split the parts of the Program.cs into files just like in Asset Service

File: src/Lykke.Snow.Notifications/appsettings.dev.json (hunk 2 above)
Comment (fatihdumanli) [reply src/Lykke.Snow.Notifications/appsettings.dev.json]: That's my bad, now I removed. I also updated .gitignore

File: tests/Lykke.Snow.Notifications.Tests/Lykke.Snow.Notifications.Tests.csproj (hunk 3 above)
Comment (fatihdumanli) [reply tests/Lykke.Snow.Notifications.Tests/Lykke.Snow.Notifications.Tests.csproj]: Upgraded

**Review by tarurar — APPROVED** [review]

**Pull request merged**